webanno-io-csv/src/main/java/de/tudarmstadt/ukp/clarin/webanno/csv/WebAnnoCsvReader.java

```diff
@@ -47,31 +52,54 @@ public void getNext(JCas aJCas) throws IOException, CollectionException {
                     .parse(new InputStreamReader(is, "UTF-8"));
             List<String> headers = new ArrayList<>();
             for (CSVRecord record : records) {
-                String documentName = record.get(0);
-                String annotator = record.get(1);
-
                 if (headers.isEmpty()) {
+                    String documentName = record.get(0);
+                    String annotator = record.get(1);
+
                     headers.add(documentName);
                     headers.add(annotator);
                     for (int c = 2; c < record.size(); c++) {
                         headers.add(record.get(c));
                     }
-                } else {
-                    String text = record.get((int) (record.size() - 1));
+                }
+                else {
+                    String text = record.get(record.size() - 1);
                     if (null != text && null == aJCas.getDocumentText()) {
                         aJCas.setDocumentText(text);
                     }
+                    // add the codebook annotations
+                    // TODO this needs to get generified for every type of (custom) annotation if we
+                    // TODO want CSV support not only for Codebook
+                    createCodebookAnnotations(aJCas, headers, record);
 
                 }
 
             }
-        } catch (Exception e) {
+        }
+        catch (Exception e) {
             throw new CollectionException(e);
-        } finally {
+        }
+        finally {
             closeQuietly(is);
 
         }
 
     }
 
+    private void createCodebookAnnotations(JCas aJCas, List<String> headers, CSVRecord record)
+        throws CollectionException
+    {
+        AnnotationFS codebookAnnotation;
+        for (int i = 2; i < record.size() - 1; i++) {
```

Comment:
2 seems to be a magic number here so there should be a comment here explaining why to start at 2.